src/worker/compile_query.ts

```diff
@@ -20,7 +20,7 @@ export const compileQuery = async (
   const {uri} = documentMeta;
   const url = new URL(uri);
   const connectionLookup = connectionManager.getConnectionLookup(url);
-  const runtime = new Runtime(fileHandler, connectionLookup);
+  const runtime = new Runtime({ urlReader: fileHandler, connections: connectionLookup });
```

Comment:
Hmm seems like we should be using the compile cache here too

Reply:
Yeah, probably.

Reply:
The short answer for why it doesn't is that the worker and the LSP have remained logically separate for the most part even though they now inhabit the same process for various technical reasons. But since it's been over a year since we last reversed a decision on that it may be time to eliminate the artificial separation.